content/exchange/artifacts/Server.Monitor.Flows.Windows.EventLogs.Hayabusa.yaml

```diff
@@ -0,0 +1,52 @@
+name: Server.Monitor.Flows.Windows.EventLogs.Hayabusa
+author: Whitney Champion - bluesky @whit.zip, Eric Capuano - bluesky @eric.zip
+description: |
+   This server monitoring artifact will automatically collect any KAPE triage artifacts, and send them for processing with Hayabusa.
+
+type: SERVER_EVENT
+
+parameters:
+ - name: ArtifactNameRegex
+   default: "Windows.KapeFiles.Targets"
+   description: A regular expression to select which artifacts to upload
+ - name: VelociraptorDataDir
+   default:
+
+sources:
+ - name: Process
+   query: |
+
+    -- Get flow completions for hunt/collection
+    LET completions = SELECT *, client_info(client_id=ClientId).os_info.fqdn AS Fqdn 
+        FROM watch_monitoring(artifact="System.Flow.Completion")
+        WHERE Flow.artifacts_with_results =~ ArtifactNameRegex
+        
+    -- Get flows and file data for each flow 
+    LET flows = SELECT * FROM foreach(
+        row=completions, 
+        query={
+            SELECT ClientId, *, Data["VFSPath"] AS VFSPath FROM enumerate_flow(client_id=ClientId, flow_id=FlowId) LIMIT 1
+        }
+    )
+    
+    -- Get location of evtx files to send to hayabusa 
+    LET evtx_files = SELECT * FROM foreach(
+        row=flows,
+        query={
+            SELECT ClientId, OSPath FROM stat(filename=VFSPath, accessor="fs") WHERE OSPath =~ ".evtx" LIMIT 1
+        }
+    )
+    
+    -- Get directory
+    LET directories = SELECT ClientId, parse_string_with_regex(string=OSPath, regex='fs:((/clients/[^/]+/collections/[^/]+/uploads/auto/C:/Windows/System32/winevt/)[^/]+/)') AS Directory FROM evtx_files
```

Comment:
You can use file_store() to get the backing file directly

https://docs.velociraptor.app/vql_reference/server/file_store/